src/main/java/com/renatusnetwork/momentum/gameplay/listeners/JoinLeaveListener.java

```diff
@@ -194,6 +200,52 @@ public void onQuit(PlayerQuitEvent event) {
             playerStats.resetLevel();
         }
 
+        Squad squad = playerStats.getSquad();
+        if (playerStats.getSquad() != null) {
+            boolean leader = SquadsManager.isLeader(playerStats);
+            squadsManager.leave(playerStats);
+            squadsManager.addOffline(playerStats.getUUID(), squad);
+            if (squad.size() == 0) { // everyone leaves
+                squadsManager.disband(squad);
+            } else {
+                if (leader) {
+                    SquadsManager.notifyMembers(squad, "&3The squad leader, &9" + playerStats.getDisplayName() + "&3, has gone offline! They have one minute to rejoin before leadership changes");
+                    new BukkitRunnable() {
```

Comment:
I feel like we can divide this up into helper functions to make it simpler and less duplicate code